src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,56 +1,139 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './FindMovie.scss';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+import { Loader } from '../Loader';
+
+type Props = {
+  setMovies: (movies: Movie[]) => void;
+  movies: Movie[];
+};
+
+export const FindMovie: React.FC<Props> = ({ setMovies, movies }) => {
+  const [movie, setMovie] = useState<Movie | undefined>();
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleFindMovie = async (event: React.SyntheticEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    const movieFromServer = await getMovie(query.trim());
+
+    if (movieFromServer.Response === 'False') {
+      setError(true);
+    } else {
+      setError(false);
+      setMovie({
+        title: movieFromServer.Title,
+        description: movieFromServer.Plot,
+        imgUrl: movieFromServer.Poster === 'N/A'
+          ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+          : movieFromServer.Poster,
+        imdbUrl: `https://www.imdb.com/title/${movieFromServer.imdbID}`,
+        imdbId: movieFromServer.imdbID,
+      });
+    }
+
+    setIsLoading(false);
+  };
+
+  const handleAddMovie = () => {
+    if (movie) {
+      if (movies.find(m => m.imdbId === movie.imdbId)) {
+        setQuery('');
+        setMovie(undefined);
+        // eslint-disable-next-line no-alert
+        alert('This movie is already in the list');
+
+        return;
+      }
+
+      setMovies([
+        ...movies,
+        movie,
+      ]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setMovies.